Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanViewModel.kt

```diff
@@ -74,6 +88,25 @@ class SubmissionQRCodeScanViewModel @AssistedInject constructor(
         }
     }
 
+    // Collect Test result registration only after user has given a consent.
+    // To exclude any registered test result before giving a consent
+    private suspend fun saveTestResultAnalyticsSettings(testResult: TestResult) = with(analyticsSettings) {
```

Comment:
Can you move this logic to dedicated class as it has nothing to do with the screen logic?

Reply:
I did not get it. Where do you want me to put? it is in the ViewModel , Does the ViewModel handles only screen logic ?

Reply:
Improved ,Moved the logic of saving to `TestResultDonorSettings` ,but it still entry point is here. As per tech specs this should happen in QR scan, if you know a better handling let me know.

Reply:
You could just move the whole method to a dedicated class and inject that one instead.

Reply:
Done ✅, new class , new method , new test coverage